Source/WebKit/UIProcess/API/Swift/WebPage.swift

```diff
@@ -24,13 +24,80 @@
 #if ENABLE_SWIFTUI
 
 import Foundation
+import Observation
+import SwiftUI // FIXME: Do not import SwiftUI in WebKit proper.
 
 @_spi(Internal)
-@MainActor public class WebPage_v0 {
+@MainActor
+@Observable
+public class WebPage_v0 {
     public let navigations: Navigations
 
+    public var url: URL? {
+        accessBackingProperty(\.url, backedBy: \.url)
+    }
+
+    public var title: String {
+        // The title property is annotated as optional in WKWebView, but is never actually `nil`.
+        _$observationRegistrar.access(self, keyPath: \.title)
+        return backingWebView.title!
+    }
+
+    public var estimatedProgress: Double {
+        accessBackingProperty(\.estimatedProgress, backedBy: \.estimatedProgress)
+    }
+
+    public var isLoading: Bool {
+        accessBackingProperty(\.isLoading, backedBy: \.isLoading)
+    }
+
+    public var serverTrust: SecTrust? {
+        accessBackingProperty(\.serverTrust, backedBy: \.serverTrust)
+    }
+
+    public var hasOnlySecureContent: Bool {
+        accessBackingProperty(\.hasOnlySecureContent, backedBy: \.hasOnlySecureContent)
+    }
+
+    public var themeColor: Color? {
+        // The color property is a UIColor/NSColor in WKWebView.
+        _$observationRegistrar.access(self, keyPath: \.themeColor)
+        return backingWebView.themeColor.map(Color.init(_:))
```

Comment:
Aren't the unlabeled conversion initializer to-be-deprecated[^1][^2]? There are also `init(nsColor:)`/`init(uiColor:)` versions.

[^1]: https://developer.apple.com/documentation/swiftui/color/init(_:)-9rl8m
[^2]: https://developer.apple.com/documentation/swiftui/color/init(_:)-1uwrw